### ISSUE TITLE
Problem about maxpooling kernel size and strides

### ISSUE THREAD

**Issue by yueruchen** [peace195/sppnet#1]
The implementation of  kernel size and strides of maxpooling in your code is this:
```
 for i in range(len(out_pool_size)):
        h_strd = previous_conv_size[0] / out_pool_size[i]
        w_strd = previous_conv_size[1] / out_pool_size[i]
        h_wid = previous_conv_size[0] - h_strd * out_pool_size[i] + 1
        w_wid = previous_conv_size[1] - w_strd * out_pool_size[i] + 1
        max_pool = tf.nn.max_pool(previous_conv,
                                   ksize=[1,h_wid,w_wid, 1],
                                   strides=[1,h_strd, w_strd,1],
                                   padding='VALID')
        if (i == 0):
            spp = tf.reshape(max_pool, [num_sample, -1])
        else:
            spp = tf.concat(axis=1, values=[spp, tf.reshape(max_pool, [num_sample, -1])])
```

But the official caffe code is this:
![image](https://user-images.githubusercontent.com/14892792/31922789-ab2c1f08-b83c-11e7-86f4-3badbc1e358a.png)

**Comment by yueruchen** [comment]
Can you give a reason about your implementation?
Thanks!

**Comment by peace195** [comment]
I am too busy to day. I will feedback to you as soon as possible.

Thank you very much.

Binh Thanh Do

2017-10-24 9:54 GMT+07:00 Yifan Jiang <notifications@github.com>:

**Comment by peace195** [comment]
@yueruchen I developed the tensorflow version of SPP followed by the idea in the [paper](https://arxiv.org/pdf/1406.4729.pdf).

Thank you for your comparison. Let me clear something here:

* I use floor() instead of ceil() function for stride window size.

* I do not use padding in the SPP layer.

I have updated my code. Could you check it? Thank you very much.

**Comment by WanliOuyang** [comment]
I've run your code, but the accuracy of training and testing is very low.
